From: Simon McVittie <smcv@collabora.com>
Date: Mon, 18 Mar 2019 17:34:00 +0000
Subject: build-recipe-dsc: Show information about the build chroot

This vaguely mimics what Debian's sbuild does, although the precise
information given here is different.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Forwarded: https://github.com/openSUSE/obs-build/pull/497
---
 build-recipe-dsc | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/build-recipe-dsc b/build-recipe-dsc
index c4fe3e2..428a737 100644
--- a/build-recipe-dsc
+++ b/build-recipe-dsc
@@ -89,6 +89,31 @@ dsc_build() {
     local changes
     local f
 
+    echo "==== Kernel ===="
+    echo
+    uname -a || true
+    echo
+
+    echo "==== Filesystems in build chroot ===="
+    echo
+    chroot "$buildroot" su -c "cat /proc/self/mountinfo" - "$BUILD_USER" < /dev/null || true
+    echo
+
+    echo "==== User identity in build chroot ===="
+    echo
+    chroot "$buildroot" su -c "id" - "$BUILD_USER" < /dev/null || true
+    echo
+
+    echo "==== Environment variables in build chroot ===="
+    echo
+    chroot "$buildroot" su -c "env | LC_ALL=C sort -u" - "$BUILD_USER" < /dev/null || true
+    echo
+
+    echo "==== Packages installed in build chroot ===="
+    echo
+    chroot "$buildroot" su -c "dpkg-query -W" - "$BUILD_USER" < /dev/null || true
+    echo
+
     DSC_BUILD_OPTIONS=
     if test -n "$BUILD_JOBS" ; then
 	DSC_BUILD_OPTIONS="parallel=${BUILD_JOBS}"
